File: frontend/src/types/collection.ts
export interface CollectionRead {
  id: number
  name: string
  source_folder: string
  created_at: string
  item_count: number
}

export interface CollectionCreate {
  name: string
  source_folder: string
}

export interface IngestionStatus {
  processed: number
  total: number
  stage: 'idle' | 'scanning' | 'ingesting' | 'done'
}

File: backend/src/app/schemas/collection.py
from datetime import datetime

from pydantic import BaseModel


class CollectionCreate(BaseModel):
    """Payload for creating a new collection."""

    name: str
    source_folder: str


class CollectionRead(BaseModel):
    """Collection data returned to the client."""

    id: int
    name: str
    source_folder: str
    created_at: datetime
    item_count: int
